src/database/inmemory/js-generator.ts

```diff
@@ -859,6 +860,12 @@ register(FieldPathQueryNode, (node, context) => {
     return fragment;
 });
 
+register(FlexSearchStartsWithQueryNode, (node, context) => {
+    const lhs = processNode(node.lhs, context);
+    const rhs = processNode(node.rhs, context);
+    return js`(${lhs}.startsWith(${rhs}))`;
```

Comment:
this is missing a null check on `lhs`. In `BinaryOperationQueryNode`, I used js`String(${lhs}).startsWith(${rhs})`. You could add a test where you use `_starts_with` on a field that has null values somewhere (e.g. `Country.totalInvestment`).

Reply:
Country.totalInvestment doesn't work, because of permissions. I will add a new field to delivery that is null in some cases.